src/model/data/ERC20.ts

```diff
@@ -1,109 +1,36 @@
 import { IntentAction } from '../IntentAction';
+import { Function } from './Function';
 import BigNumber = require("bn.js");
 
-const Web3 = require('web3');
-const web3 = new Web3();
-
 export class ERC20 {
     contractAddress: string;
 
     constructor(contractAddress: string) {
```

Comment:
Maybe we can move the constructor to its own class, and inherit contract implementations from there